File: code/taskflow-task/src/main/java/com/taskflow/task/entity/Task.java
package com.taskflow.task.entity;

import com.taskflow.common.entity.AuditEntity;
import com.taskflow.task.constant.enums.Priority;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
@Table(name = "tasks")
public class Task extends AuditEntity {

    @Column(name = "project_id", nullable = false)
    private Long projectId;

    @Column(name = "board_id", nullable = false)
    private Long boardId;

    @Column(name = "list_id", nullable = false)
    private Long listId;

    @Column(name = "sprint_id")
    private Long sprintId;

    @Column(nullable = false, length = 500)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "assignee_id")
    private Long assigneeId;

    @Column(name = "reporter_id", nullable = false)
    private Long reporterId;

    @Column(name = "due_date")
    private Long dueDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Priority priority = Priority.MEDIUM;

    @Column(nullable = false)
    private Integer position = 0;

    @Version
    @Column(nullable = false)
    private Long version;
}

File: code/taskflow-fe/src/types/task.ts
export type Priority = 'LOW' | 'MEDIUM' | 'HIGH' | 'URGENT';

export interface Task {
  id: number;
  project_id: number;
  board_id: number;
  list_id: number;
  sprint_id?: number | null;
  title: string;
  description?: string | null;
  assignee_id?: number | null;
  reporter_id?: number | null;
  due_date?: number | null;
  priority?: Priority | null;
  position: number;
  version: number;
  created_at: number;
  last_updated_at: number;
  label_ids?: number[];
}

export interface CreateTaskRequest {
  list_id: number;
  title: string;
  description?: string;
  assignee_id?: number;
  due_date?: number;
  priority?: Priority;
  label_ids?: number[];
  sprint_id?: number;
}

export interface UpdateTaskRequest {
  title?: string;
  description?: string;
  assignee_id?: number | null;
  due_date?: number | null;
  priority?: Priority | null;
  label_ids?: number[];
  sprint_id?: number | null;
  version: number;
}

export interface MoveTaskRequest {
  to_list_id: number;
  position?: number;
}

export interface Label {
  id: number;
  project_id: number;
  name: string;
  color: string;
}

export interface CreateLabelRequest {
  project_id: number;
  name: string;
  color: string;
}
